pkg/plugins/tsdb.go

```diff
@@ -189,6 +190,44 @@ type DataResponse struct {
 	Message string                     `json:"message,omitempty"`
 }
 
+// ToBackendDataResponse converts the legacy format to the standard SDK format
+func (r DataResponse) ToBackendDataResponse() (*backend.QueryDataResponse, error) {
+	qdr := &backend.QueryDataResponse{
+		Responses: make(map[string]backend.DataResponse, len(r.Results)),
+	}
+
+	// Convert tsdb results (map) to plugin-model/datasource (slice) results.
+	// Only error, Series, and encoded Dataframes responses are mapped.
+	for refID, res := range r.Results {
+		pRes := backend.DataResponse{}
+		if res.Error != nil {
+			pRes.Error = res.Error
+		}
+
+		if res.Dataframes != nil {
+			decoded, err := res.Dataframes.Decoded()
+			if err != nil {
+				return qdr, err
+			}
+			pRes.Frames = decoded
+			qdr.Responses[refID] = pRes
+			continue
+		}
+
+		for _, series := range res.Series {
+			frame, err := SeriesToFrame(series)
+			frame.RefID = refID
+			if err != nil {
+				return nil, err
+			}
```

Comment:
It's a bad idea to access the non-error return value in case of an error, it could cause a nil pointer exception (depending on whether it's a pointer or not). It also makes reading difficult.
```suggestion
			frame, err := SeriesToFrame(series)
			if err != nil {
				return nil, err
			}
			frame.RefID = refID
```

Reply:
thanks!  this code was just moved -- i'll fix